### ISSUE TITLE
[QUESTION]: Can .NET interactive be embedded in a Web Application? 

### ISSUE THREAD

**Issue by rrekapalli** [dotnet/interactive#800]
Hi there,

Does this library be embedded in a web application developed in ASP.NET/Angular/React? 

Thanks

**Comment by jonsequitur** [comment]
Yes. There are a number of packages on NuGet and you can see an embedding example in [this sample](https://github.com/dotnet/interactive/tree/main/samples/connect-wpf).

**Comment by colombod** [comment]
We are working on exposing the tools to configure your aspnet host with .NET Interactive in PR #810 802

**Comment by rrekapalli** [comment]
Thank you!

**Comment by PawelStadnicki** [comment]
@colombod aspnet host PR mentioned above seems to be merged now however the is no sample yet/blog/tutorial?
Hosting dotnet-interactive on custom webpage is highly anticipated IMHO

**Comment by colombod** [comment]
> @colombod aspnet host PR mentioned above seems to be merged now however there is no sample yet/blog/tutorial about it?
> Hosting dotnet-interactive on custom webpage is highly anticipated IMHO

Consuming .ET Interactive from a web page is not hard to begin but crafting a good experience takes time. We do not have yet documentation or tutorial for it but you can take a peak at a simple demo I have built [here](https://github.com/colombod/dotnet-interactive-experiments/tree/main/dotnet-interactive-html)

**Comment by PawelStadnicki** [comment]
Thanks, I will try it.
Will it be supported to use other kernels too ? I guess that GitHub Codespaces has the same functionality as VS Code. 
Wonder how much effort will be required to host on custom page same functionality as VS Code or GitHub Codespaces ( I assume that VS Code & Github Codespaces uses internal "notebook" support that must be adjusted for own purposes).

I can also guess that definitely there will be some blazor-flavour at some point.
I also know that MS internally uses some "trydotnetjs" that can also be public at some point
but grasping it all is hard without the full roadmap.

**Comment by colombod** [comment]
> Thanks, I will try it.
> Will it be supported to use other kernels too ? I guess that GitHub Codespaces has the same functionality as VS Code.
> Wonder how much effort will be required to host on custom page same functionality as VS Code or GitHub Codespaces ( I assume that VS Code & Github Codespaces uses internal "notebook" support that must be adjusted for own purposes).
> 
> I can also guess that definitely there will be some blazor-flavour at some point.
> I also know that MS internally uses some "trydotnetjs" that can also be public at some point
> but grasping it all is hard without the full roadmap.

You can already use any other kernel for .NET Interactive. That app has the same engine as the VSCode, Jupiter, and Jupiterlab integration. You can access the sharp, fsharp, powershell and js languages. To have same functionality as the code spaces you might want to look at the [nteract SDK](https://github.com/nteract/nteract) to create your frontend. Wiring up [Monaco editor](https://github.com/microsoft/monaco-editor) could be complex but I would not be surprised if the nteract ecosystem already has a Monaco integration
